### PR TITLE
virtcontainers/fc: Upgrade Firecracker to v0.23.1

### PR DESCRIPTION
**Pull request by olagkasn**
This patch upgrades Firecracker version from v0.21.1 to v0.23.1

* Generate swagger models for v0.23.1 (from firecracker.yaml)
* Change uint64 types in TokenBucket object according to rate-limiter
implementation (introduced in commit #cfeb966)
* Update Firecracker Logger/Metrics to support the new API
* Update payload in fc.vmRunning to support the new API
* Add Metrics type to fcConfig

Fixes: #1518

Signed-off-by: Orestis Lagkas Nikolos <olagkasn@nubificus.co.uk>

### CODE REVIEW

File: src/runtime/virtcontainers/fc.go [hunk 1]
```diff
@@ -611,19 +599,29 @@ func (fc *firecracker) fcSetLogger() error {
 		return fmt.Errorf("Failed setting log: %s", err)
 	}
 
-	// listen to metrics file and transfer error info
+        fc.fcConfig.Logger = &models.Logger{
```
Comment (liubin) [inline src/runtime/virtcontainers/fc.go:602]: This file is not formatted?

Comment (olagkasn) [reply src/runtime/virtcontainers/fc.go]: hm ok, I force pushed after running gofmt

**Comment by olagkasn** [comment]
so does anyone have an idea about the failing checks? I don't see any errors when building it on our machines

**Comment by fidencio** [comment]
```
# github.com/kata-containers/kata-containers/src/runtime/virtcontainers
../virtcontainers/fc.go:612:21: not enough arguments in call to fc.trace
	have (string)
	want ("context".Context, string)
Makefile:604: recipe for target '/tmp/jenkins/workspace/kata-containers-2.0-clh-crio-ubuntu-PR/go/src/github.com/kata-containers/kata-containers/src/runtime/kata-runtime' failed
make: *** [/tmp/jenkins/workspace/kata-containers-2.0-clh-crio-ubuntu-PR/go/src/github.com/kata-containers/kata-containers/src/runtime/kata-runtime] Error 2
Failed at 59: make
Failed at 38: "${cidir}/install_runtime.sh" "${tag}"
```

Did it change after the trace work done by @cmaf?  Most likely.  This PR would need a rebase & a quick build / testing atop of `main` and then should be good to go.

**Comment by olagkasn** [comment]
> ```
> # github.com/kata-containers/kata-containers/src/runtime/virtcontainers
> ../virtcontainers/fc.go:612:21: not enough arguments in call to fc.trace
> 	have (string)
> 	want ("context".Context, string)
> Makefile:604: recipe for target '/tmp/jenkins/workspace/kata-containers-2.0-clh-crio-ubuntu-PR/go/src/github.com/kata-containers/kata-containers/src/runtime/kata-runtime' failed
> make: *** [/tmp/jenkins/workspace/kata-containers-2.0-clh-crio-ubuntu-PR/go/src/github.com/kata-containers/kata-containers/src/runtime/kata-runtime] Error 2
> Failed at 59: make
> Failed at 38: "${cidir}/install_runtime.sh" "${tag}"
> ```
> 
> Did it change after the trace work done by @cmaf? Most likely. This PR would need a rebase & a quick build / testing atop of `main` and then should be good to go.

done :)

**Comment by fidencio** [comment]
The current error is **not** related to this PR, and seems like an infra issue that should be solved soon on softwares that we rely on:
```
12:33:57 Submodule 'meson' (https://git.qemu.org/git/meson.git) registered for path 'meson'
12:33:58 Submodule 'tests/fp/berkeley-softfloat-3' (https://git.qemu.org/git/berkeley-softfloat-3.git) registered for path 'tests/fp/berkeley-softfloat-3'
12:33:58 Submodule 'tests/fp/berkeley-testfloat-3' (https://git.qemu.org/git/berkeley-testfloat-3.git) registered for path 'tests/fp/berkeley-testfloat-3'
12:33:58 Submodule 'ui/keycodemapdb' (https://git.qemu.org/git/keycodemapdb.git) registered for path 'ui/keycodemapdb'
12:33:58 Cloning into '/root/qemu/meson'...
12:33:58 error: RPC failed; curl 18 transfer closed with outstanding read data remaining
12:33:58 fatal: the remote end hung up unexpectedly
12:35:51 fatal: clone of 'https://git.qemu.org/git/meson.git' into submodule path '/root/qemu/meson' failed
12:35:51 Failed to clone 'meson'. Retry scheduled
12:35:51 Cloning into '/root/qemu/tests/fp/berkeley-softfloat-3'...
12:35:51 fatal: unable to access 'https://git.qemu.org/git/berkeley-softfloat-3.git/': gnutls_handshake() failed: Error in the pull function.
12:35:51 fatal: clone of 'https://git.qemu.org/git/berkeley-softfloat-3.git' into submodule path '/root/qemu/tests/fp/berkeley-softfloat-3' failed
12:36:37 Failed to clone 'tests/fp/berkeley-softfloat-3'. Retry scheduled
12:36:37 Cloning into '/root/qemu/tests/fp/berkeley-testfloat-3'...
12:36:37 error: RPC failed; curl 18 transfer closed with outstanding read data remaining
12:36:37 fatal: the remote end hung up unexpectedly
12:39:01 fatal: protocol error: bad pack header
12:39:01 fatal: clone of 'https://git.qemu.org/git/berkeley-testfloat-3.git' into submodule path '/root/qemu/tests/fp/berkeley-testfloat-3' failed
12:39:01 Failed to clone 'tests/fp/berkeley-testfloat-3'. Retry scheduled
12:39:01 Cloning into '/root/qemu/meson'...
12:39:01 fatal: unable to access 'https://git.qemu.org/git/meson.git/': Empty reply from server
12:39:01 fatal: clone of 'https://git.qemu.org/git/meson.git' into submodule path '/root/qemu/meson' failed
12:39:33 Failed to clone 'meson' a second time, aborting
12:39:33 ./scripts/git-submodule.sh: failed to update modules
```

**Review by fidencio — APPROVED** [review]
lgtm, thanks @olagkasn!

**Pull request merged**